v3/lints/rfc/lint_ext_authority_key_identifier_no_key_identifier.go

```diff
@@ -57,7 +59,23 @@ func (l *authorityKeyIdNoKeyIdField) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *authorityKeyIdNoKeyIdField) Execute(c *x509.Certificate) *lint.LintResult {
-	if c.AuthorityKeyId == nil && !util.IsSelfSigned(c) { //will be nil by default if not found in x509.parseCert
+	if util.IsCACert(c) && util.IsSelfSigned(c) {
```

Comment:
Maybe we can remove util.IsCACert(c)? Only if it is self-signed it may be omitted.

Reply:
I'm not quite convinced of that based on the language.

> There is one exception;  where a CA distributes its public key in the form of a "self-signed" certificate, the authority key identifier MAY be omitted.

It seems to me that this clause is stating that this does apply, specifically to CA certs and not just to any self signed.

Reply:
That is fine.